### PR TITLE
[TRTLLM-15078][test] Prune non-Llama-3.1-8B Llama tests

### PR DESCRIPTION
**Pull request by xinhe-nv**
## Dev Engineer Review

- Pruned non-Llama-3.1-8B model references, accuracy data, performance configurations, and integration tests.
- Replaced selected TinyLlama usage with Llama-3.1-8B-Instruct.
- Removed Ray, MPI, LoRA, KV-cache, LLM API, and disaggregated-serving coverage.
- Added Blackwell SM120 Mixtral E2E coverage and Qwen3 Eagle3 coverage.
- Validate removed functional coverage, model paths, YAML files, test references, waiver entries, and `test_llm.py` golden outputs.
- No production runtime changes were identified.

## QA Engineer Review

Test code changed across accuracy, disaggregated serving, E2E, KV-cache, LLM API, Ray, MPI, LoRA, sampler, and AutoDeploy areas.

Removed or modified tests include accuracy classes, disaggregated-serving functions, Ray and MPI examples, KV-cache tests, LLM API tests, LoRA helpers, sampler tests, and model-path fixtures. Several retained tests now use Llama-3.1-8B-Instruct.

Affected manifests:

- `tests/integration/test_lists/qa/llm_function_core.txt`
- `tests/integration/test_lists/qa/llm_spark_core.txt`
- `tests/integration/test_lists/qa/llm_spark_func.yml`
- `tests/integration/test_lists/qa/llm_spark_perf.yml`
- `tests/integration/test_lists/test-db/l0_a10.yml`
- `tests/integration/test_lists/test-db/l0_a100.yml`
- `tests/integration/test_lists/test-db/l0_b200.yml`
- `tests/integration/test_lists/test-db/l0_dgx_b200.yml`
- `tests/integration/test_lists/test-db/l0_dgx_h100.yml`
- `tests/integration/test_lists/test-db/l0_dgx_h200.yml`
- `tests/integration/test_lists/test-db/l0_gb200_multi_gpus.yml`
- `tests/integration/test_lists/test-db/l0_h100.yml`
- `tests/integration/test_lists/test-db/l0_l40s.yml`
- `tests/integration/test_lists/test-db/l0_sanity_check.yml`
- `tests/integration/test_lists/waives.txt`

CBTS coverage data is unavailable. Verdict: **needs follow-up**.

## Description

The background is https://docs.google.com/spreadsheets/d/17qLRr5oQESqn1rT9LrIUnpjvhH8wcC59273YEPT9k44/edit?gid=177162358#gid=177162358

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by xinhe-nv** [comment]
@coderabbitai review the change, we need to retire llama tests, if the test is must have, please show me in a table.

File: tests/integration/defs/disaggregated/test_configs/disagg_config_overlap_transceiver_runtime_python_bounce.yaml [hunk 1]
```diff
@@ -1,49 +0,0 @@
-# Disaggregated config that exercises the KV-cache bounce buffer.
```
Comment (BowenFu) [inline tests/integration/defs/disaggregated/test_configs/disagg_config_overlap_transceiver_runtime_python_bounce.yaml:1]: This file is deleted here, but #17334 (Kimi K3 disagg) *modifies* it in the same window — both are open against `main`. Whichever lands second either conflicts or silently drops the other's intent. Worth syncing with @brnguyen2 on whether the bounce config stays and moves to a surviving model, or #17334 should stop depending on it.

File: tests/integration/defs/disaggregated/test_disaggregated.py [hunk 2]
```diff
@@ -1010,744 +1006,6 @@ def run_disaggregated_test(example_dir,
...
-    run_disaggregated_test(disaggregated_example_root,
-                           "conditional",
-                           env=llm_venv._new_env,
-                           model_path=llama_model_root,
-                           cwd=llm_venv.get_working_directory())
-
-
-@pytest.mark.parametrize("llama_model_root", ['TinyLlama-1.1B-Chat-v1.0'],
-                         indirect=True)
-def test_disaggregated_ngram(disaggregated_test_root, llm_venv,
```
Comment (BowenFu) [inline tests/integration/defs/disaggregated/test_disaggregated.py:1803]: Most of what's removed here has a DeepSeek-V3-Lite analog that survives (`ctxpp2_gentp2`, `ctxtp2ep2pp2_gentp4`, `transceiver_runtime_python`, `overlap*`, `cache_aware_balance`, `conditional`), so the prune reads as deliberate. But these look like they leave **no** surviving coverage at all, on any model:

- `test_disaggregated_ngram` — ngram speculative decoding under disagg
- `test_disaggregated_load_balance` — plain round-robin balancing (only `cache_aware_balance` survives)
- `test_disaggregated_perf_metrics` and `test_disaggregated_kv_cache_time_output`
- `test_disaggregated_chat_completion_tool_calls`
- `test_disaggregated_python_transceiver_host_offload`, `..._bounce`, `..._fabric_memory`
- `test_disaggregated_overlap_gen_first`
- the ctx/gen PP x TP topologies `ctxpp2_genpp2`, `ctxtp2_genpp2`, `ctxtp2pp2_gentp2pp2`, `ctxpp4_genpp4`, `ctxpp4_gentp4`

These are *feature* and *topology* coverage that happened to run on TinyLlama rather than Llama coverage. Could you confirm each of those is either covered elsewhere or intentionally dropped? Anything in the second bucket would be better as a port onto DSv3-Lite in this PR than as a gap nobody notices until it regresses.

Comment (xinhe-nv) [reply tests/integration/defs/disaggregated/test_disaggregated.py:1803]: should we use other models to cover these scenarios? for example gpt-oss-20b or qwen3-0.6b, because DSv3-Lite is also a obsoleted model

File: tests/integration/defs/disaggregated/test_configs/disagg_config_overlap_transceiver_runtime_python_bounce.yaml (hunk 1 above)
Comment (xinhe-nv) [reply tests/integration/defs/disaggregated/test_configs/disagg_config_overlap_transceiver_runtime_python_bounce.yaml:1]: Kimi K3 is p0 model, I can keep the pr until https://github.com/NVIDIA/TensorRT-LLM/pull/17334 is merged.

**Comment by Shixiaowei02** [comment]
The changes to the dis-agg testing logic are quite extensive, and I'd like to spend some more time reviewing them thoroughly.

**Comment by chuangz0** [comment]
The tests in test_ray.py and the disaggregated-serving test suite are not TinyLlama accuracy tests. Many of them provide functional coverage and should be preserved by migrating them to a maintained model, such as Qwen3-0.6B or Llama 3.1 8B.

File: tests/integration/defs/disaggregated/test_disaggregated.py (hunk 2 above)
Comment (BowenFu) [reply tests/integration/defs/disaggregated/test_disaggregated.py:1803]: Yes — that's the right shape, and @chuangz0's point is the same one. I'd split it by what the test is actually for:

**Topology/mechanism coverage** (`ctxpp2_genpp2`, `ctxtp2_genpp2`, `ctxtp2pp2_gentp2pp2`, `ctxpp4_genpp4`, `ctxpp4_gentp4`, `multi_orchestrator`, `router`, `cuda_graph`, `mixed`, `genbs1`, `diff_max_tokens`) is model-agnostic — Qwen3-0.6B is the cheapest home for it and porting is mostly a config swap.

**Feature coverage** is the part I'd not lose: `ngram`, `load_balance`, `perf_metrics`, `kv_cache_time_output`, `chat_completion_tool_calls`, `python_transceiver_host_offload`, `..._bounce`, `..._fabric_memory`, `overlap_gen_first(_pp4)`, `benchmark_gen_only(_insufficient_kv)`. Each of those is the only test exercising its code path, so dropping them removes the signal entirely rather than moving it.

`llama4_long_context_kv_cache_overflow` is the one that genuinely needs a big model, so it can't just move to Qwen3-0.6B.

No objection to the Llama-accuracy pruning itself — it's this second list I'd want landed somewhere before this merges.

File: tests/unittest/_torch/sampler/test_best_of_n.py [hunk 3]
```diff
@@ -34,8 +34,8 @@ def expected_outputs():
 
 @pytest.fixture(scope="module")
 def llm():
-    return LLM(model=os.path.join(llm_models_root(), "llama-models-v2",
-                                  "TinyLlama-1.1B-Chat-v1.0"),
+    return LLM(model=os.path.join(llm_models_root(), "llama-3.1-model",
+                                  "Llama-3.1-8B-Instruct"),
```
[the remark this answers is not in the document]
Comment (xinhe-nv) [reply tests/unittest/_torch/sampler/test_best_of_n.py:38]: The finding is not valid. expected_outputs is only compared when n == best_of (line 110), but the parametrize gives n=2 with best_of as None or 3 — so n == best_of is always False. The fixture's snapshot strings are dead code and regenerating them would have no effect on test behavior. The model path at lines 37–38 is already llama-3.1-model/Llama-3.1-8B-Instruct, which is correct.

File: tests/unittest/llmapi/test_llm.py [hunk 4]
```diff
@@ -124,7 +120,7 @@ def llm_check_output(llm: LLM,
                  stop_reasons=stop_reasons)
 
 
-default_model_name = "llama-models-v2/TinyLlama-1.1B-Chat-v1.0"
+default_model_name = "llama-3.1-model/Llama-3.1-8B-Instruct"
 mixtral_model_name = "Mixtral-8x7B-v0.1"
 
 llama_model_path = get_model_path(default_model_name)
```
[the remark this answers is not in the document]
Comment (xinhe-nv) [reply tests/unittest/llmapi/test_llm.py:126]: None of the three sub-items are "still-valid issues" (broken code needing a fix); they are feature additions. Finding is SKIP.

File: tests/integration/defs/disaggregated/test_disaggregated.py [hunk 5]
```diff
@@ -1718,8 +1582,9 @@ def test_disaggregated_kv_cache_time_output(disaggregated_test_root, llm_venv,
     env["TRTLLM_KVCACHE_TIME_OUTPUT_PATH"] = output_path
     run_disaggregated_test(disaggregated_example_root,
                            "perf_metrics",
-                           env=env,
-                           model_path=llama_model_root,
+                           env=llm_venv._new_env
```
Comment (BowenFu) [inline tests/integration/defs/disaggregated/test_disaggregated.py]: This drops the UCX/C++ backend variables assembled above, so the test no longer guarantees it exercises the C++ timing CSV path.
```suggestion
                           env=env
```

File: tests/integration/defs/disaggregated/test_disaggregated.py [hunk 6]
```diff
@@ -1128,128 +1139,98 @@ def test_disaggregated_single_gpu(disaggregated_test_root,
...
 
-def _verify_mamba_bs1_concurrency2(server_url: str) -> None:
-
-    async def run() -> None:
-        timeout = aiohttp.ClientTimeout(total=120)
-        prompts = (
-            "Write one sentence about disaggregated inference.",
-            "Write one sentence about recurrent-state transfer.",
-        )
+@pytest.mark.parametrize("llama_model_root", ['Qwen3-0.6B'], indirect=True)
```
Comment (BowenFu) [inline tests/integration/defs/disaggregated/test_disaggregated.py:1138]: This force-push deletes `test_disaggregated_mamba_bs1_concurrency2`, which covers recurrent-state transfer rather than a retired Llama model. Could you restore that test?

File: tests/integration/test_lists/waives.txt [hunk 7]
```diff
@@ -74,9 +74,7 @@ accuracy/test_llm_api_pytorch.py::TestLlama3_1_8BInstruct::test_fp8_4gpus[pp4-fp
 accuracy/test_llm_api_pytorch.py::TestLlama3_1_8BInstruct::test_fp8_4gpus[tp2pp2-fp8kv=False-attn_backend=TRTLLM-torch_compile=False] SKIP (https://nvbugs/6427411)
 accuracy/test_llm_api_pytorch.py::TestLlama3_1_8BInstruct::test_fp8_4gpus[tp2pp2-fp8kv=True-attn_backend=TRTLLM-torch_compile=False] SKIP (https://nvbugs/6427411)
 accuracy/test_llm_api_pytorch.py::TestLlama3_1_8BInstruct::test_guided_decoding_4gpus[xgrammar] SKIP (https://nvbugs/6427411)
-accuracy/test_llm_api_pytorch.py::TestLlama3_3_70BInstruct::test_fp4_tp2pp2[torch_compile=False-enable_gemm_allreduce_fusion=False] SKIP (https://nvbugs/6428089)
-accuracy/test_llm_api_pytorch.py::TestLlama3_3_70BInstruct::test_fp4_tp2pp2[torch_compile=True-enable_gemm_allreduce_fusion=False] SKIP (https://nvbugs/6427411)
-accuracy/test_llm_api_pytorch.py::TestLlama3_3_70BInstruct::test_fp4_tp2pp2[torch_compile=True-enable_gemm_allreduce_fusion=True] SKIP (https://nvbugs/6211441)
+accuracy/test_llm_api_pytorch.py::TestMiniMaxM2::test_4gpus[attention_dp=False-cuda_graph=True-overlap_scheduler=True-tp_size=4-ep_size=4] SKIP (https://nvbugs/6159132)
```
Comment (BowenFu) [inline tests/integration/test_lists/waives.txt]: This waiver is unrelated to retiring or migrating Llama tests. Please split it into the owning fix or waiver change.
```suggestion
```

File: tests/integration/test_lists/waives.txt [hunk 8]
```diff
@@ -182,6 +171,7 @@ full:B300/accuracy/test_llm_api_pytorch.py::TestMistralLarge3_675B::test_nvfp4_4
 full:B300/accuracy/test_llm_api_pytorch.py::TestNemotronV3Ultra::test_nvfp4_8gpus[attention_dp_off-trtllm] SKIP (https://nvbugs/6474894)
 full:B300/accuracy/test_llm_api_pytorch.py::TestQwen3_30B_A3B::test_dummy_load_format SKIP (https://nvbugs/6525059)
 full:B300/accuracy/test_llm_api_pytorch.py::TestStep3_7::test_fp8_block_scales[tp_size=4-ep_size=4-mtp_nextn=3] SKIP (https://nvbugs/6539941)
+full:B300/disaggregated/test_auto_scaling.py::test_service_discovery[http-kv_cache_aware] SKIP (https://nvbugs/6474892)
```
Comment (BowenFu) [inline tests/integration/test_lists/waives.txt]: This waiver is unrelated to retiring or migrating Llama tests. Please split it into the owning fix or waiver change.
```suggestion
```

File: tests/integration/test_lists/qa/llm_spark_perf.yml [hunk 9]
```diff
@@ -66,11 +62,9 @@ llm_spark_perf:
         gte: 2
         lte: 2
   tests:
-  - perf/test_perf.py::test_perf[llama_v3.3_70b_instruct-bench-pytorch-streaming-bfloat16-maxbs:1-input_output_len:2048,128-reqs:1-con:1-tp:2-gpus:2]
   - perf/test_perf.py::test_perf[qwen3_235b_a22b_fp4-bench-pytorch-streaming-float4-maxbs:1-input_output_len:2048,128-kv_cache_dtype:fp8-reqs:1-con:1-tp:2-gpus:2]
   - perf/test_perf.py::test_perf[qwen3_235b_a22b_fp4-bench-pytorch-streaming-float4-maxbs:1-input_output_len:128,2048-kv_cache_dtype:fp8-reqs:1-con:1-tp:2-gpus:2]
-  - perf/test_perf.py::test_perf[llama_v4_scout_17b_16e_instruct_fp8-bench-pytorch-streaming-float8-maxbs:1-input_output_len:2048,128-kv_cache_dtype:fp8-reqs:1-con:1-ep:2-tp:2-gpus:2]
-  - perf/test_perf.py::test_perf[llama_v4_scout_17b_16e_instruct-bench-pytorch-streaming-bfloat16-maxbs:1-input_output_len:2048,128-reqs:1-con:1-tp:2-gpus:2]
+  - perf/test_perf.py::test_perf[deepseek_r1_distill_llama_70b-bench-pytorch-streaming-bfloat16-maxbs:1-input_output_len:2048,128-reqs:1-con:1-tp:2-gpus:2]
```
Comment (xinhe-nv) [inline tests/integration/test_lists/qa/llm_spark_perf.yml]: delete the test

File: tensorrt_llm/_torch/pyexecutor/sampler/sampler_strategy.py [hunk 10]
```diff
@@ -292,43 +292,43 @@ def sample(
         case ("top_k", top_k, temperature):
             tokens, softmax = top_k_top_p_sampling_batch(
                 logits,
-                top_k=cast(int, top_k),
-                temperature=cast(float, temperature),
+                top_k=cast(int, top_k),  # type: ignore[redundant-cast]
```
Comment (BowenFu) [inline tensorrt_llm/_torch/pyexecutor/sampler/sampler_strategy.py]: Could you drop this unrelated sampler-typing commit from the test-pruning PR? #17601 already carries the focused class-pattern fix; keeping `redundant-cast` suppressions here widens scope and creates a competing implementation.

File: pyproject.toml [hunk 11]
```diff
@@ -955,6 +955,8 @@ disable_error_code = [
     # fragile WAR. Suppress redundant-cast for the sampler module until the casts are
     # audited and the annotations tightened, rather than churn the source.
     "redundant-cast",
+    # temperature is typed as float but set to None in the greedy case (sampler_strategy.py).
+    "assignment",
 ]
```
Comment (BowenFu) [inline pyproject.toml:958]: Could you remove this sampler-wide `assignment` suppression and its preceding comment from this test-pruning PR? It weakens future checks across the module; keep the focused type fix in the owning sampler change.
